lib/Telemetry/src/DispatcherFactory.h

```diff
@@ -0,0 +1,114 @@
+#include "DataQueue.h"
+#include "LogCommand.h"
+#include "Command.h"
+#include "Dispatcher.h"
+#include "Particle.h"
+#include "IntervalLogger.h"
+
+#define MAX_NUM_INTERVALS 5
+
+#ifndef _DISPTACHER_FACTORY_H_
+#define _DISPATCHER_FACTORY_H_
+
+// This isn't really a factory but it has a similar purpose:
+// Add all the commands you want to this and call build().  It
+// produces a dispatcher with all logging functionality already set up
+// You can delete this safely after build() has been called
+class DispatcherFactory {
+    public:
+        /**
+         * Constructs new DispatcherFactory with number of commands and data queue
+         * 
+         * @param numCommands The number of sensor functions you will add to the factory
+         * Note: this must be accurate and has potential to cause a segfault if the value is too low
+         * @param dataQ the Data Queue from main
+         * */
+        DispatcherFactory(uint16_t numCommands, DataQueue *dataQ) {
+            _dataQ = dataQ;
+            _commands = new Command*[numCommands];
```

Comment:
Does _commands ever get deleted? I'm not seeing it in the code, but maybe I'm just not looking hard enough.

Reply:
I think I wasn't sure if deleting _commands would also delete the objects that _command's pointers point to.  I'm going to look this up now because that wouldn't really make sense

Reply:
I've set it up to be deleted now